magento/inventory
# MC-21944: Unable to save a credit memo

### Description (*)

When the product SKUs are numeric-only (even if presented as a string type) and used as array keys
```
inventory/app/code/Magento/InventorySourceSelectionApi/Model/Algorithms/Result/GetDefaultSortedSourcesResult.php:98
```
PHP casts those keys to integers forcibly and there is no control over it. As a result, we are getting key => value pairs where keys (SKUs) are integers. Later along the way, this causes an issue when using a _searchCriteriaBuilder_ here
```
inventory/app/code/Magento/InventorySourceSelectionApi/Model/GetInStockSourceItemsBySkusAndSortedSource.php:53
```
which receives SKUs as integers and performing the following SQL query
```
SELECT `main_table`.* FROM `inventory_source_item` AS `main_table` WHERE ((`sku` IN(11001))) AND ((`source_code` IN('default'))) AND ((`status` = 1));
```
while SKU is a VARCHAR column. Because of that, the above query returns both 1101 and 1101TW products (from the example product SKUs of the Manual Testing Section) into the results, hence an exception is thrown after that.

Converting SKUs to string fixes the issue described.

### Fixed Issues (if relevant)

1. magento/magento2#MC-21944: Unable to save a credit memo

### Manual testing scenarios (*)

1. Create 2 simple products with SKUs 1101 and 1101TW
2. Place an order using 1101 SKU
3. Try to create a credit memo for the order
You will see "The credit memo couldn't be saved" error

### Questions or comments


### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
 - [*] All commits are accompanied by meaningful commit messages
 - [*] All new or changed code is covered with unit/integration tests (if applicable)
 - [*] All automated tests passed successfully (all builds are green)
